modules/kvtagger/kvtagger.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2016 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "kvtagger.h"
+#include "csv-tagrecord-scanner.h"
+#include "logmsg/logmsg.h"
+#include "logpipe.h"
+#include "parser/parser-expr.h"
+#include "reloc.h"
+#include "tagrecord-scanner.h"
+#include "template/templates.h"
+#include "kvtagdb.h"
+
+#include <stdio.h>
+#include <string.h>
+
+
+typedef struct KVTagger
+{
+  LogParser super;
+  KVTagDB *tagdb;
+  gchar *selector_template_string;
+  LogTemplate *selector_template;
+  gchar *default_selector;
+  gchar *filename;
+  gchar *prefix;
+} KVTagger;
+
+void
+kvtagger_set_filename(LogParser *p, const gchar *filename)
+{
+  KVTagger *self = (KVTagger *)p;
+
+  g_free(self->filename);
+  self->filename = g_strdup(filename);
+}
+
+void
+kvtagger_set_database_selector_template(LogParser *p, const gchar *selector)
+{
+  KVTagger *self = (KVTagger *)p;
+
+  g_free(self->selector_template_string);
+  self->selector_template_string = g_strdup(selector);
+}
+
+void
+kvtagger_set_prefix(LogParser *p, const gchar *prefix)
+{
+  KVTagger *self = (KVTagger *)p;
+
+  g_free(self->prefix);
+  self->prefix = g_strdup(prefix);
+}
+
+void
+kvtagger_set_database_default_selector(LogParser *p, const gchar *default_selector)
+{
+  KVTagger *self = (KVTagger *)p;
+
+  g_free(self->default_selector);
+  self->default_selector = g_strdup(default_selector);
+}
+
+gboolean
```

Comment:
If we consider what this function does and where it is used, I don't feel the added value of keeping it.